### PR TITLE
Excess whitespace removal throughout a string

### PR DESCRIPTION
**Pull request by stephlocke**
Function uses `str_replace()` and `stri_trim_both()` to remove excess spaces.

### CODE REVIEW

**Comment by stephlocke** [comment]
@hadley - new PR with requested stuff from #177

File: NEWS.md [hunk 1]
```diff
@@ -12,6 +12,8 @@
 * `str_remove()` and `str_remove_all()` functions. These wrap 
   `str_replace()` and `str_replace_all()` to remove patterns from strings.
   (@Shians, #178)
+  
+* `str_squish()` removes excess spaces from both the left and right side of strings. (@stephlocke #197)
```
Comment (hadley) [inline NEWS.md]: And the middle?

Comment (stephlocke) [reply NEWS.md]: Good spot 

Added: b171a5e

**Comment by hadley** [comment]
Thanks!

**Pull request merged**